### PR TITLE
Refactor NetCDF Find Module

### PR DESCRIPTION
**Pull request by program--**
This PR refactors the NetCDF find module to correctly use components. The current module does not set component found vars.

Now, when configuring NGen, we should correctly see:

```
...

-- Found NetCDF: /usr/include (found version "4.9.2") found components: CXX

...
```

## Changes

- Overhaul of `cmake/FindNetCDF.cmake`.

## Todos

- [x] Add module documentation

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows project standards (link if applicable)
- [x] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [X] New functions are documented (with a description, list of inputs, and expected output)
- [X] Placeholder code is flagged / future todos are captured in comments
- [X] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [x] Linux
- [x] macOS

### CODE REVIEW

**Comment by PhilMiller** [comment]
This may be mooted or simplified by #729 if that's embraced.

File: CMakeLists.txt [hunk 1]
```diff
@@ -182,10 +182,7 @@ endif()
 
 # -----------------------------------------------------------------------------
 if(NGEN_WITH_NETCDF)
-    set(NETCDF_CXX "YES")
-    set(NETCDF_F77 "NO")
-    set(NETCDF_F90 "NO")
-    find_package(NetCDF REQUIRED)
+    find_package(NetCDF REQUIRED COMPONENTS CXX)
```
Comment (donaldwj) [inline CMakeLists.txt:185]: Should probably require the C and CXX components at a minimum.

File: cmake/FindNetCDF.cmake [hunk 2]
```diff
@@ -1,162 +1,146 @@
...
+
+# =============================================================================
+include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(NetCDF
-  DEFAULT_MSG NETCDF_LIBRARIES NETCDF_INCLUDE_DIRS NETCDF_HAS_INTERFACES NETCDF_VERSION)
-
-function(FindNetCDF_get_is_parallel_aware include_dir)
-  file(STRINGS "${include_dir}/netcdf_meta.h" _netcdf_lines
-    REGEX "#define[ \t]+NC_HAS_PARALLEL[ \t]")
-  string(REGEX REPLACE ".*NC_HAS_PARALLEL[ \t]*([0-1]+).*" "\\1" _netcdf_has_parallel "${_netcdf_lines}")
```
Comment (donaldwj) [inline cmake/FindNetCDF.cmake:141]: Not sure if we want to loose parallel netcdf detection. I was planning on using parallel output for the netcdf writer.

**Review by donaldwj — CHANGES_REQUESTED** [review]
The main issue I see is this seems to loose parallel netcdf detection. If possible I would like that capability kept, If not simple I will approve as is.

File: CMakeLists.txt (hunk 1 above)
Comment (program--) [reply CMakeLists.txt:185]: The C library isn't a component, it's the base module, so if you call `find_package(NetCDF)` that only gets the C library.

File: cmake/FindNetCDF.cmake (hunk 2 above)
Comment (program--) [reply cmake/FindNetCDF.cmake:141]: Parallel detection is on lines 52-60.

Comment (program--) [reply cmake/FindNetCDF.cmake:141]: https://github.com/NOAA-OWP/ngen/blob/b59d96ac5960d202cb0f9aa1a30fa3dffe9faf7d/cmake/FindNetCDF.cmake#L51-L60

Comment (donaldwj) [reply cmake/FindNetCDF.cmake:141]: Sorry I missed git was showing your entire file as a change and I did not find the string when I searched.

**Review by donaldwj — APPROVED** [review]

**Pull request merged**